### PR TITLE
fix(machines): repeated machine.count API calls

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- Itemised list of what was changed by this PR.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to machine list
- Verify that only 2 machine.count API calls are made when there are no search filters
- Select a filter
- Verify that a single additional single call has been made for the count of machines matching the selected filter

## Fixes

Fixes: https://github.com/canonical/maas-ui/issues/4488

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.

### CODE REVIEW

**Comment by ndv99** [comment]
Is this ready for review or is it still in draft?

**Comment by petermakowski** [comment]
> Is this ready for review or is it still in draft?

Ready for review now.

**Review by ndv99 — APPROVED** [review]
Looks good!

**Comment by ndv99** [comment]
@sparkiegeek has requested that we hold off on merging this, awaiting his approval to merge.

**Pull request merged**
